index.js

```diff
@@ -0,0 +1,167 @@
+const express = require('express');
+
+//Can remove all session related items ahead of configing tokens
+
+// const session = require('express-session');
+
+// const KnexSessionStore = require('connect-session-knex')(session);
+
+const jwt = require('jsonwebtoken');
+
+const cors = require('cors');
+
+const db = require('./database/dbConfig.js');
+
+const bcrypt = require('bcryptjs');
+
+const server = express();
+
+const cookieTime = (req, res, next) => {
+    const token = req.headers.authorization;
+
+    if(token) {
+        jwt.verify(token, jwtSecret, (err, decodedToken) => {
+            if(err) {
+                res.status(401).json({ message: "Invalid toke, Take another"})
+            } else {
+                req.decodedToken = decodedToken;
+                next();
+            }
+        })
+    }
+    else {
+        res.status(401).json({message: 'Take a toke.'})
+    }
+
+    // next();
+//   if (req.session && req.session.username) {
+//     next();
+//     } else {
+//       res.status(401).send('Not authorized');
+//     }
+  }
+
+
+// const sessionConfig = {
+//     name:'notsession', ///we want to change to anon session software
+//     secret:'something%dwarf#something',
+//     cookie: { 
+//       secure: false,
+//       maxAge: 1000 * 60 * 1
+//     },
+//     httpOnly: true, ///block JS
+//     resave: false,
+//     saveUninitialized: false,
+//     store: new KnexSessionStore({
+//       tablename: 'sessions',
+//       sidefieldname: 'sid',
+//       knex: db,
+//       createtable: true,
+//       clearIntervale: 1000 * 60 * 60,
+//     }),
+// };
+
+
+const jwtSecret = 'quikbrownfox';
+
+function generateToken(user) {
+    
+    const jwtPayload = { 
+        ...user,
+        hello: 'FSW13',
+        role: 'admin'
+    };
+
+    // const jwtSecret = 'quikbrownfox';
+
+    const jwtOptions = {
+        expiresIn: '1m',
+    };
+    
+    return jwt.sign(jwtPayload, jwtSecret, jwtOptions)
+};
+
+server.use(
+  express.json(), 
+  cors(),
+//   session(sessionConfig),
+);
+
+server.post('/api/register', (req, res) => {
+  const credentials = req.body; 
+
+  const hash = bcrypt.hashSync(credentials.password, 14);
+  credentials.password = hash;
+
+  db('users').insert(credentials).then(ids => {
+    const id = ids[0];
+    // req.session.username = credentials.username;  
+    res.status(201).json({ newUserId: id})
+  })
+   .catch(err => {
+     res.status(500).json(err);
```

Comment:
Write an error message explaining why the user couldn't register and how they can solve the issue.